internal/cmd/beta/network-interface/describe/describe.go

```diff
@@ -0,0 +1,181 @@
+package describe
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/goccy/go-yaml"
+	"github.com/spf13/cobra"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/args"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/errors"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/examples"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/flags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/globalflags"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/print"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/services/iaas/client"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/tables"
+	"github.com/stackitcloud/stackit-cli/internal/pkg/utils"
+	"github.com/stackitcloud/stackit-sdk-go/services/iaas"
+)
+
+const (
+	nicIdArg = "NIC_ID"
+
+	networkIdFlag = "network-id"
+)
+
+type inputModel struct {
+	*globalflags.GlobalFlagModel
+	NetworkId *string
+	NicId     string
+}
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "describe",
+		Short: "Describes a network interface",
+		Long:  "Describes a network interface.",
+		Args:  args.SingleArg(nicIdArg, utils.ValidateUUID),
+		Example: examples.Build(
+			examples.NewExample(
+				`Describes network interface with nic id "xxx" and network ID "yyy"`,
+				`$ stackit beta network-interface describe xxx --network-id yyy`,
+			),
```

Comment:
You could also include an example using the `--output` flag, to maintain consistency with other commands